app/lib/service/entrypoint/_isolate.dart

```diff
@@ -138,9 +141,23 @@ Future startIsolates({
     });
     logger.info('Worker isolate #$workerIndex started.');
 
+    // Automatically killing worker isolate between in the next ~48 hours.
+    //
+    // This clears us from temporary processing hangups, at most within 2 days,
+    // probably in a few hours (as we have many workers and probably a few will
+    // be triggered earlier).
+    final ttl =
+        Duration(hours: 24 + _random.nextInt(24), minutes: _random.nextInt(60));
+    logger.info('Worker isolate #$workerIndex TTL set to $ttl.');
+    final autoKillTimer = Timer(ttl, () {
+      logger.info('Killing worker isolate #$workerIndex...');
+      isolate.kill();
```

Comment:
Won't this cause some analysis tasks to fail? How long before they are retried?

That's what I meant by terminating gracefully.

Reply:
> Won't this cause some analysis tasks to fail? How long before they are retried?

Good point. I've updated the code the include an alive-callback mechanism, where the worker will send regular pings to the controller, and that resets the timer. While working, it won't forcefully restart the isolate.